### PR TITLE
simplify test_block_store

### PR DESCRIPTION
**Pull request by arvidn**
use the `DBConnection` utility context manager to set up and tear down a test database and `DBWrapper` object.

best reviewed with white-space changes disabled

### CODE REVIEW

File: tests/core/full_node/test_block_store.py [hunk 1]
```diff
@@ -30,60 +28,38 @@ async def test_block_store(self, tmp_dir, db_version):
...
+                    await store.set_peak(block_record.header_hash)
+                    await store.set_peak(block_record.header_hash)
+
+                assert len(await store.get_full_blocks_at([1])) == 1
+                assert len(await store.get_full_blocks_at([0])) == 1
+                assert len(await store.get_full_blocks_at([100])) == 0
+
+                # Get blocks
+                block_record_records = await store.get_block_records_in_range(0, 0xFFFFFFFF)
+                assert len(block_record_records) == len(blocks)
```
Comment (xdustinface) [inline tests/core/full_node/test_block_store.py:62]:
```suggestion
    async with DBConnection(db_version) as db_wrapper, DBConnection(db_version) as db_wrapper_2:
        # Use a different file for the blockchain
        coin_store_2 = await CoinStore.create(db_wrapper_2)
        store_2 = await BlockStore.create(db_wrapper_2)
        hint_store = await HintStore.create(db_wrapper_2)
        bc = await Blockchain.create(coin_store_2, store_2, test_constants, hint_store, tmp_dir)

        store = await BlockStore.create(db_wrapper)
        await BlockStore.create(db_wrapper_2)

        # Save/get block
        for block in blocks:
            await bc.receive_block(block)
            block_record = bc.block_record(block.header_hash)
            block_record_hh = block_record.header_hash
            await store.add_full_block(block.header_hash, block, block_record)
            await store.add_full_block(block.header_hash, block, block_record)
            assert block == await store.get_full_block(block.header_hash)
            assert block == await store.get_full_block(block.header_hash)
            assert block_record == (await store.get_block_record(block_record_hh))
            await store.set_peak(block_record.header_hash)
            await store.set_peak(block_record.header_hash)

        assert len(await store.get_full_blocks_at([1])) == 1
        assert len(await store.get_full_blocks_at([0])) == 1
        assert len(await store.get_full_blocks_at([100])) == 0

        # Get blocks
        block_record_records = await store.get_block_records_in_range(0, 0xFFFFFFFF)
        assert len(block_record_records) == len(blocks)
```

File: tests/core/full_node/test_block_store.py [hunk 2]
```diff
@@ -93,42 +69,29 @@ async def test_deadlock(self, tmp_dir, db_version):
...
+                    rand_i = random.randint(0, 9)
+                    if random.random() < 0.5:
+                        tasks.append(
+                            asyncio.create_task(
+                                store.add_full_block(blocks[rand_i].header_hash, blocks[rand_i], block_records[rand_i])
+                            )
+                        )
+                    if random.random() < 0.5:
+                        tasks.append(asyncio.create_task(store.get_full_block(blocks[rand_i].header_hash)))
+                await asyncio.gather(*tasks)
```
Comment (xdustinface) [inline tests/core/full_node/test_block_store.py:97]:
```suggestion
        async with DBConnection(db_version) as wrapper, DBConnection(db_version) as wrapper_2:
            store = await BlockStore.create(wrapper)
            coin_store_2 = await CoinStore.create(wrapper_2)
            store_2 = await BlockStore.create(wrapper_2)
            hint_store = await HintStore.create(wrapper_2)
            bc = await Blockchain.create(coin_store_2, store_2, test_constants, hint_store, tmp_dir)
            block_records = []
            for block in blocks:
                await bc.receive_block(block)
                block_records.append(bc.block_record(block.header_hash))
            tasks = []

            for i in range(10000):
                rand_i = random.randint(0, 9)
                if random.random() < 0.5:
                    tasks.append(
                        asyncio.create_task(
                            store.add_full_block(blocks[rand_i].header_hash, blocks[rand_i], block_records[rand_i])
                        )
                    )
                if random.random() < 0.5:
                    tasks.append(asyncio.create_task(store.get_full_block(blocks[rand_i].header_hash)))
            await asyncio.gather(*tasks)
```

**Review by xdustinface — CHANGES_REQUESTED** [review]
This

```
async with X() as x1:
    async with X() as x2:
    ....
```

can be transformed into

```
async with X() as x1, X() as x2:
    ....
```

unless i miss something.

**Comment by arvidn** [comment]
@xdustinface Updated. I didn't take your patches because I couldn't see them with whitespace disabled. I'm sure it's identical though

**Comment by arvidn** [comment]
I'm a bit surprised I didn't have to say `async` twice.. so presumably I can't mix an async context manager with a normal one

**Review by xdustinface — APPROVED** [review]

**Pull request merged**